src/pcre2_compile_class.c

```diff
@@ -958,21 +991,23 @@ case-independence.
 
 Arguments:
   options       the options bits
+  xoptions      the extra options bits
   cb            contains pointers to tables etc.
   p             points to row of 32-bit values, terminated by NOTACHAR
 
 Returns:        cb->classbits is updated
 */
 
 static void
-add_list_to_class(uint32_t options, compile_block *cb, const uint32_t *p)
+add_list_to_class(uint32_t options, uint32_t xoptions, compile_block *cb,
```

Comment:
As far as I remember this code is only used for pre-defined bitsets, such as:
https://github.com/PCRE2Project/pcre2/blob/master/src/pcre2_compile_class.c#L1319

All of them removes caseless. Maybe we could just pass 0 as xoptions, and don't change the interface.
Minor optimization: maybe the caseless removal could be done in this function as well: `options &= ~PCRE2_CASELESS`.